### PR TITLE
Lint:fix lints

### PR DESCRIPTION
**Pull request by Thor-wl**
Try to address issue #791
* add some golangci-lint rules
* fix some items found by golangci-lint

### CODE REVIEW

File: pkg/client/clientset/versioned/typed/scheduling/v1beta1/scheduling_client.go [hunk 1]
```diff
@@ -46,9 +46,7 @@ func (c *SchedulingV1beta1Client) Queues() QueueInterface {
 // NewForConfig creates a new SchedulingV1beta1Client for the given config.
 func NewForConfig(c *rest.Config) (*SchedulingV1beta1Client, error) {
 	config := *c
-	if err := setConfigDefaults(&config); err != nil {
-		return nil, err
-	}
+	setConfigDefaults(&config)
```
Comment (k82cn) [inline pkg/client/clientset/versioned/typed/scheduling/v1beta1/scheduling_client.go:49]: those are generated code, so please do not modify them.

File: pkg/scheduler/actions/preempt/preempt.go [hunk 2]
```diff
@@ -185,7 +185,7 @@ func preempt(
 	stmt *framework.Statement,
 	preemptor *api.TaskInfo,
 	filter func(*api.TaskInfo) bool,
-) (bool, error) {
+) bool {
```
Comment (k82cn) [inline pkg/scheduler/actions/preempt/preempt.go:188]: let's keep such kind of error which may be used later.

File: pkg/scheduler/actions/reclaim/reclaim.go [hunk 3]
```diff
@@ -50,7 +50,7 @@ func (ra *Action) Execute(ssn *framework.Session) {
 	klog.V(3).Infof("There are <%d> Jobs and <%d> Queues in total for scheduling.",
 		len(ssn.Jobs), len(ssn.Queues))
 
-	var underRequest []*api.JobInfo
+	// var underRequest []*api.JobInfo
```
Comment (k82cn) [inline pkg/scheduler/actions/reclaim/reclaim.go:53]: why remove this?

File: pkg/scheduler/cache/event_handlers.go [hunk 4]
```diff
@@ -246,18 +246,16 @@ func (sc *SchedulerCache) DeletePod(obj interface{}) {
...
 // Assumes that lock is already acquired.
-func (sc *SchedulerCache) addNode(node *v1.Node) error {
+func (sc *SchedulerCache) addNode(node *v1.Node) {
 	if sc.Nodes[node.Name] != nil {
 		sc.Nodes[node.Name].SetNode(node)
 	} else {
 		sc.Nodes[node.Name] = schedulingapi.NewNodeInfo(node)
 	}
-
-	return nil
```
Comment (k82cn) [inline pkg/scheduler/cache/event_handlers.go:256]: let's keep this.

File: pkg/scheduler/cache/event_handlers.go [hunk 5]
```diff
@@ -286,11 +284,7 @@ func (sc *SchedulerCache) AddNode(obj interface{}) {
 	sc.Mutex.Lock()
 	defer sc.Mutex.Unlock()
 
-	err := sc.addNode(node)
-	if err != nil {
-		klog.Errorf("Failed to add node %s into cache: %v", node.Name, err)
-		return
-	}
+	sc.addNode(node)
```
Comment (k82cn) [inline pkg/scheduler/cache/event_handlers.go:287]: let's keep oldNode, if not used, try `_`.

**Comment by k82cn** [comment]
please do not change the signature of function in this PR.

File: .golangci.yml [hunk 6]
```diff
@@ -103,3 +102,10 @@ issues:
         - golint
         - deadcode
         - godot
+
+linters-settings:
+  lll:
+    line-length: 256
+  funlen:
```
Comment (hzxuzhonghu) [inline .golangci.yml:109]: I think we can disable this check. In the following pr, we need to first enable that with high priority. 

As to the priority, If we have no enough time to analyze I suggest looking at https://github.com/istio/istio/blob/494aca43715d2f970bdb76ce1dfcc5e1095f3070/common/config/.golangci.yml#L34-L53

File: pkg/scheduler/actions/reclaim/reclaim.go (hunk 3 above)
Comment (hzxuzhonghu) [reply pkg/scheduler/actions/reclaim/reclaim.go:53]: It is useless

File: pkg/scheduler/actions/reclaim/reclaim.go [hunk 7]
```diff
@@ -64,22 +64,18 @@ func (ra *Action) Execute(ssn *framework.Session) {
...
+			queues.Push(queue)
 		}
 
 		if len(job.TaskStatusIndex[api.Pending]) != 0 {
 			if _, found := preemptorsMap[job.Queue]; !found {
 				preemptorsMap[job.Queue] = util.NewPriorityQueue(ssn.JobOrderFn)
 			}
 			preemptorsMap[job.Queue].Push(job)
-			underRequest = append(underRequest, job)
+			// underRequest = append(underRequest, job)
```
Comment (hzxuzhonghu) [inline pkg/scheduler/actions/reclaim/reclaim.go:78]: please remove directly

File: pkg/client/clientset/versioned/typed/scheduling/v1beta1/scheduling_client.go (hunk 1 above)
Comment (Thor-wl) [reply pkg/client/clientset/versioned/typed/scheduling/v1beta1/scheduling_client.go]: done

File: pkg/scheduler/actions/reclaim/reclaim.go (hunk 3 above)
Comment (Thor-wl) [reply pkg/scheduler/actions/reclaim/reclaim.go:53]: useless variable has been already deleted

File: pkg/scheduler/actions/reclaim/reclaim.go (hunk 7 above)
Comment (Thor-wl) [reply pkg/scheduler/actions/reclaim/reclaim.go:78]: done

File: pkg/scheduler/cache/event_handlers.go (hunk 4 above)
Comment (Thor-wl) [reply pkg/scheduler/cache/event_handlers.go:256]: recoveried

File: pkg/scheduler/cache/event_handlers.go (hunk 5 above)
Comment (Thor-wl) [reply pkg/scheduler/cache/event_handlers.go:287]: recoveried

File: .golangci.yml (hunk 6 above)
Comment (Thor-wl) [reply .golangci.yml]: OK

**Pull request merged**